packages/features/ee/workflows/components/WorkflowStepContainer.tsx

```diff
@@ -504,717 +569,660 @@ export default function WorkflowStepContainer(props: WorkflowStepProps) {
-                      render={({ field: { value, onChange } }) => (
-                        <PhoneInput
-                          placeholder={t("phone_number")}
-                          id={`steps.${step.stepNumber - 1}.sendTo`}
-                          className="min-w-fit sm:rounded-r-none sm:rounded-bl-md sm:rounded-tl-md"
-                          required
+              {numberVerified ? (
+                <div className="mt-1">
+                  <Badge variant="green">{t("number_verified")}</Badge>
+                </div>
+              ) : (
+                !props.readOnly && (
+                  <>
+                    <div className="mt-6 flex w-full flex-col">
+                      <Label className="">{t("verification_code")}</Label>
+                      <div className="flex w-full items-center">
+                        <TextField
+                          containerClassName="w-full"
+                          className="h-8 rounded-xl"
+                          placeholder={t("code")}
                           disabled={props.readOnly}
-                          value={value}
-                          onChange={(val) => {
-                            const isAlreadyVerified = !!verifiedNumbers
-                              ?.concat([])
-                              .find((number) => number.replace(/\s/g, "") === val?.replace(/\s/g, ""));
-                            setNumberVerified(isAlreadyVerified);
-                            onChange(val);
+                          value={verificationCode}
+                          onChange={(e) => {
+                            setVerificationCode(e.target.value);
                           }}
+                          required
                         />
+                        <Button
+                          color="secondary"
+                          size="sm"
+                          className="ml-2 min-w-fit rounded-xl"
+                          disabled={verifyPhoneNumberMutation.isPending || props.readOnly}
+                          onClick={() => {
+                            verifyPhoneNumberMutation.mutate({
+                              phoneNumber: form.getValues(`steps.${step.stepNumber - 1}.sendTo`) || "",
+                              code: verificationCode,
+                              teamId,
+                            });
+                          }}>
+                          {t("verify")}
+                        </Button>
+                      </div>
+                    </div>
+                    {form.formState.errors.steps &&
+                      form.formState?.errors?.steps[step.stepNumber - 1]?.sendTo && (
+                        <p className="text-error mt-1 text-xs">
+                          {form.formState?.errors?.steps[step.stepNumber - 1]?.sendTo?.message || ""}
+                        </p>
                       )}
+                  </>
+                )
```

Comment:
![Image](https://github.com/user-attachments/assets/393bc134-5ea7-4d2a-b973-9c61684f9499)

too much space between checkbox and label